staging/src/k8s.io/kube-apiextensions-server/pkg/apiserver/customresource_handler.go

```diff
@@ -353,13 +360,38 @@ func (s UnstructuredNegotiatedSerializer) SupportedMediaTypes() []runtime.Serial
 }
 
 func (s UnstructuredNegotiatedSerializer) EncoderForVersion(serializer runtime.Encoder, gv runtime.GroupVersioner) runtime.Encoder {
-	return unstructured.UnstructuredJSONScheme
+	return versioning.NewDefaultingCodecForScheme(Scheme, crEncoderInstance, nil, gv, nil)
 }
 
 func (s UnstructuredNegotiatedSerializer) DecoderToVersion(serializer runtime.Decoder, gv runtime.GroupVersioner) runtime.Decoder {
 	return unstructured.UnstructuredJSONScheme
 }
 
+var crEncoderInstance = crEncoder{}
+
+// crEncoder *usually* encodes using the unstructured.UnstructuredJSONScheme, but if the type is one of the unversioned kinds
```

Comment:
actually the code is hardcoded for Status and WatchEvent, not for the unversioned types.